internal/spdxlicense/license.go

```diff
@@ -18,16 +18,20 @@ const (
 
 //go:generate go run ./generate
 
-func ID(id string) (value, other string, exists bool) {
-	id = strings.TrimSpace(id)
-	// ignore blank strings or the joiner
-	if id == "" || id == "AND" {
-		return "", "", false
-	}
+// ID returns the canonical license ID for the given license ID
+// Note: this function is only concerned with returning a best match of an SPDX license ID
+// SPDX Expressions will be handled by a parent package which will call this function
+func ID(id string) (value string, exists bool) {
 	// first look for a canonical license
-	if value, exists := licenseIDs[strings.ToLower(id)]; exists {
-		return value, "", exists
+	if value, exists := licenseIDs[cleanLicenseID(id)]; exists {
+		return value, exists
 	}
 	// we did not find, so treat it as a separate license
-	return "", id, true
+	return "", false
+}
+
+func cleanLicenseID(id string) string {
```

Comment:
Had to duplicate this between generate and the actual package. If there is a better way to share the code happy to update to that!